### PR TITLE
FEATURE : Login modal

### PR DESCRIPTION
**Pull request by sahsisunny**
**Description**
This PR introduces a new feature to enhance user experience by adding a login modal that opens when the user clicks on the "Sign In" button. The modal provides a seamless experience for users to sign in with their Google account.

**Demo**

https://github.com/Real-Dev-Squad/tiny-site-frontend/assets/70854507/9ba8db13-c175-4918-95bd-88f8f63e7ca8

**Test Coverage**
image

### CODE REVIEW

**Review by shubhamsigdar1 — APPROVED** [review]
LGTM

**Review by shubhamsigdar1 — APPROVED** [review]

**Review by gauravsinhaweb — APPROVED** [review]

File: src/components/LoginModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    useEffect(() => {
+        document.addEventListener('mousedown', handleClickOutside);
+
+        return () => {
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [onClose]);
+
+    return (
+        <div className="fixed top-0 left-0 w-full h-full flex justify-center items-center bg-gray-800 bg-opacity-50 z-50">
```
Comment (vinit717) [inline src/components/LoginModal/index.tsx:33]: Nit: you can use dialog tag to define modal

File: src/components/Navbar/index.tsx [hunk 2]
```diff
@@ -26,54 +28,68 @@ const Navbar: React.FC = () => {
...
+                                    </div>
+                                </Button>
+                            ) : (
+                                <Button
+                                    className="flex items-center space-x-2  text-white px-4 py-2 rounded-md cursor-pointer hover:bg-gray-700"
+                                    data-testid="google-login"
+                                    onClick={() => setShowLoginModal(true)}
+                                >
+                                    <span>Sign In</span>
+                                </Button>
```
Comment (vinit717) [inline src/components/Navbar/index.tsx:60]: Nit: try to avoid div and use article or section or any symantic tags

Comment (sahsisunny) [reply src/components/Navbar/index.tsx:60]: Is it acceptable to use "article" or "section" in the navigation bar? Is this considered a good practice?

File: src/components/LoginModal/index.tsx (hunk 1 above)
Comment (sahsisunny) [reply src/components/LoginModal/index.tsx:33]: Fix in this PR - #53 

[See](https://github.com/Real-Dev-Squad/tiny-site-frontend/pull/53/files#diff-2236e40afeca67fc9fbd0769eef72cc3780f421bf8ff8cbfc8c7f3e2d31c17d3R33-R54)

**Pull request merged**